### DISCUSSION TITLE
What does "timescale" mean and its function in "examples/notebooks/Getting started/Tutorial 4-Setting parameter values.ipynb"?

### DISCUSSION THREAD

**Discussion by zhengq01** [pybamm-team/PyBaMM#2081] [Q&A] [answered]
Hi there, I have a question about the function of "timescale" and "pybamm.t" in "examples/notebooks/Getting started/Tutorial 4-Setting parameter values.ipynb", and the related code is below. I cannot understand why "timescale" is needed in creating "current_interpolant", and it was generated by the previously-defined DFN model. The drive_cycle data "US06.csv" lasts for 600s, and the simulation time is set as 800s, so I am confused why the simulated result only lasts for 103.24s, which is shown in the figure below. Hope that someone can help me to understand it. Thanks a lot! 

`import pandas as pd    # needed to read the csv data file
drive_cycle = pd.read_csv("pybamm/input/drive_cycles/US06.csv", comment="#", header=None).to_numpy()  
timescale = parameter_values.evaluate(model.timescale)  ## model = pybamm.lithium_ion.DFN()
current_interpolant = pybamm.Interpolant(drive_cycle[:, 0], drive_cycle[:, 1], timescale * pybamm.t) 

parameter_values["Current function [A]"] = current_interpolant

model = pybamm.lithium_ion.SPMe()
sim = pybamm.Simulation(model, parameter_values=parameter_values)
sim.solve([0,800])
sim.plot(["Current [A]", "Terminal voltage [V]"])`

![result](https://user-images.githubusercontent.com/16679647/170903105-b8b1d496-7e6d-4da5-9b5b-d9f906fd7a29.jpg)

**Comment by valentinsulzer** [comment, marked as answer]
`timescale` is required because `pybamm.t` is dimensionless time but the interpolant is defined on dimensional time. In this case your simulation ends early because you hit the upper 4.2V cut-off. You could avoid this by increasing the "Upper voltage cut-off [V]" parameter

**Reply by valentinsulzer** [reply]
You can check why the simulation terminated by printing `solution.termination`

**Reply by zhengq01** [reply]
Thanks a lot! I got it.
